### ISSUE TITLE
Suspected false positive after upgrade: "Constant name "<xyz>" doesn't conform to UPPER_CASE naming style (invalid-name)"

### ISSUE THREAD

**Issue by ramanan-subramanian** [PyCQA/pylint#4143]
### Steps to reproduce

Given a file `test.py`:
```python
"""
pylint test
"""
CONST_LIST = ['ab', 'cd', 'ef', 'gh']
for member in CONST_LIST:
    first_letter = member[0]
    print(first_letter)
```

### Current behavior

Result of `pylint temp.py`:
```shell
************* Module test
test.py:6:4: C0103: Constant name "first_letter" doesn't conform to UPPER_CASE naming style (invalid-name)

------------------------------------------------------------------
Your code has been rated at 7.50/10 (previous run: 7.50/10, +0.00)
```

### Expected behavior

One would expect this would not raise an error here, as the `"first_letter"` is a variable to access each member of a constant inside a loop.

### pylint --version output

Result of `pylint --version` output:
```
pylint 2.7.1
astroid 2.5
Python 3.8.5 (default, Jul 28 2020, 12:59:40) 
[GCC 9.3.0]
```
I had pylint 2.5.3 and astroid 2.4.2 earlier, and the file was fine for those version.

**Comment by Pierre-Sassoulas** [comment]
I agree the variable is supposed to be in the scope of the for loop, it should not be considered a global.

**Comment by Pierre-Sassoulas** [comment]
I'm going to close as won't fix because the variable is defined in the global scope and we have another issue to authorize both style of naming for global as long as it's internally consistent that would render this moot (See https://github.com/PyCQA/pylint/issues/5677)
